### DISCUSSION TITLE
How to obtain missing documentation?

### DISCUSSION THREAD

**Discussion by olerch** [alan-turing-institute/sktime#4899] [General]
I would like to know if it is possible to get all the documentation of the estimators listed on https://www.sktime.net/en/latest/estimator_overview.html? Some documentation is missing and an error page appears when the documentation is called up.

**Comment by fkiraly** [comment]
Thanks for the report!

I wasn't aware that some of the links can be broken.

It seems to be the case that the estimator overview only links through to a full documentation page if the estimator is present somewhere in the API reference as well.

So, if the author/maintainer of the estimator forgets to add it to the API reference, the automatically generated table will link to somewhere dead.

I'll turn this into a bug report regarding the documentation page - there are a couple things imo:

* a good first issue of reporting missing documentation and/or adding them to the respective API reference page (`docs/source/api_reference`)
* an advanced testing question - as the estimator overview automatically detects all estimators, we should be able to detect the ones with dead links too
* a maintenance question about the estimator overview
* the dead links end up on `sktime-backup`, but we should be using the primary `sktime` readthedocs project.

**Reply by olerch** [reply]
Thank you for the timely response!
So is there currently another way to access the missing documentation (for instance [MCDCNNRegressor], [ResNetRegressor] or [SimpleRNNRegressor])?

**Reply by fkiraly** [reply]
> So is there currently another way to access the missing documentation

Yes, the documentation should always exist as a docstring of the respective class in the codebase.

So, 100% success ways to get to the information are:

* search for the class definition (`class ClassName`) or function definition in the codebase and inspect the docstring
* in a python interpreter, e.g., jupyter notebook, import the class and type `?ClassName`, which prints the docstring

Now, of course, some docstrings are more complete than others... but if incomplete, that's a problem with the docstring then.

The documentation build needs to be pointed manually to the class import location, which is behind the issues when you enter via `sktime.net` (that points to `readthedocs` which hosts the `sphinx` documentation builds).

**Reply by fkiraly** [reply]
I've opened a bug report here:
https://github.com/sktime/sktime/issues/4905
thanks for your observation and reporting.
